consensus/height_vote_set.go

```diff
@@ -0,0 +1,180 @@
+package consensus
+
+import (
+	"strings"
+	"sync"
+
+	. "github.com/tendermint/tendermint/common"
+	sm "github.com/tendermint/tendermint/state"
+	"github.com/tendermint/tendermint/types"
+)
+
+type RoundVoteSet struct {
+	Prevotes   *VoteSet
+	Precommits *VoteSet
+}
+
+/*
+Keeps track of all VoteSets from round 0 to round 'round'.
+
+Also keeps track of up to one RoundVoteSet greater than
```

Comment:
Oops, this comment is wrong -- the additional round is requested from consensus/state with cs.Votes.SetRound(round+1).